### PR TITLE
feat & fix: Mineable Tags for Aether Blocks

### LINKED ISSUE #252: Enhancement: Use mineable tags to determine what tools will work best on which blocks, instead of doing it through code.

### PR DESCRIPTION
**Pull request by quek04**
1.17 migrated how you determine block harvestability rules, which includes required tool tier and what tool to use on a block, to a tag based system.
PR adds Aether's blocks to these tags, and fixes the Gravitite tool ability interface to use the new way of testing for tool harvesting.

fixes #252

### CODE REVIEW

**Review by Drullkus — APPROVED** [review]

File: src/main/java/com/gildedgames/aether/common/item/tools/abilities/IGravititeToolItem.java [hunk 1]
```diff
@@ -22,23 +20,22 @@ default InteractionResult floatBlock(UseOnContext context, float destroySpeed, f
...
-//						world.addFreshEntity(entity);
-//						heldItem.hurtAndBreak(4, player, (p) -> p.broadcastBreakEvent(hand));
-//					}
-//					return InteractionResult.sidedSuccess(world.isClientSide);
-//				}
-//			}
-//		}
+		if (player != null) {
+			if ((destroySpeed == efficiency || heldItem.isCorrectToolForDrops(state)) && world.isEmptyBlock(pos.above())) {
+				if (world.getBlockEntity(pos) == null && state.getDestroySpeed(world, pos) != -1.0F) {
```
Comment (Drullkus) [inline src/main/java/com/gildedgames/aether/common/item/tools/abilities/IGravititeToolItem.java]: > `state.getDestroySpeed(world, pos) != -1.0F`

Tbh I'd just check if the speed is ` < 0f`

**Review by Drullkus — CHANGES_REQUESTED** [review]

File: src/main/java/com/gildedgames/aether/common/item/tools/abilities/IGravititeToolItem.java (hunk 1 above)
Comment (quek04) [reply src/main/java/com/gildedgames/aether/common/item/tools/abilities/IGravititeToolItem.java]: not fixed in 4f89d77

Comment (quek04) [reply src/main/java/com/gildedgames/aether/common/item/tools/abilities/IGravititeToolItem.java]: ACTUALLY fixed in 5cd736b

**Review by Drullkus — APPROVED** [review]
Wonderful

**Pull request merged**